File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_pos = {}
        for i, n in enumerate(nums):
            if target - n in num_pos.keys():
                return [i, num_pos[target-n]]
            else:
                num_pos[n] = i
        

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> n_i;
        for(int i = 0; i < nums.size(); ++i) {
            if(n_i.find(target - nums[i]) != n_i.end()) {
                return {i, n_i[target-nums[i]]};
            }
            n_i[nums[i]] = i;
        }
        return {};
    }
};